src/linear_algebra/general_linear_group.lean

```diff
@@ -84,6 +89,21 @@ begin
   exact inv_of_eq_nonsing_inv (↑A : matrix n n R),
 end
 
+/-- An element of the matrix general linear group on `(n) [fintype n]` can be considered as an
+element of the endomorphism general linear group on `n → R`. -/
+def to_linear : general_linear_group n R ≃* linear_map.general_linear_group R (n → R) :=
+units.map_equiv matrix.to_lin_alg_equiv'.to_ring_equiv.to_mul_equiv
+
+-- For some reason this doesn't trigger as a simp lemma.  Decidability issue apparently?
+-- TODO fix this and delete the special case stated for `n` rather than `fin m` below.
+lemma coe_to_linear (M : general_linear_group n R) :
+  (to_linear M : (n → R) →ₗ[R] (n → R)) = matrix.mul_vec_lin M :=
+rfl
+
+@[simp] lemma coe_to_linear' {m : ℕ} (M : general_linear_group (fin m) R) :
+  (to_linear M : (fin m → R) →ₗ[R] (fin m → R)) = matrix.mul_vec_lin M :=
+rfl
```

Comment:
If you add `general_linear_group.coe_to_linear (_ : GL (fin 2) _)` to the simp that doesn't find this, then you don't need the second lemma. I can't work out what's going on here, but it seems that lean can't deduce `n` in the first of these lemmas.

Maybe it's worth keeping both anyway, but I'd extend the comment to:
```suggestion
@[simp] lemma coe_to_linear (M : general_linear_group n R) :
  (to_linear M : (n → R) →ₗ[R] (n → R)) = matrix.mul_vec_lin M :=
rfl

/-- `matrix.general_linear_group.coe_to_linear` does not work as a simp lemma for
this case, as for some reason it is unable to deduce `n = fin m`. -/
-- TODO: fix the problem and delete this special case
@[simp] lemma coe_to_linear_fin {m : ℕ} (M : general_linear_group (fin m) R) :
  (to_linear M : (fin m → R) →ₗ[R] (fin m → R)) = matrix.mul_vec_lin M :=
rfl
```

Reply:
Aha, here's the fix!

```suggestion
-- Note that without the `@` and `‹_›`, lean infers `λ a b, _inst_1 a b` instead of `_inst_1` as the
-- decidability argument, which prevents `simp` from obtaining the instance by unification.
@[simp]
lemma coe_to_linear (M : @general_linear_group n R ‹_› ‹_› ‹_›) :
  (@to_linear n ‹_› ‹_› M : (n → R) →ₗ[R] (n → R)) = matrix.mul_vec_lin M :=
rfl
```

Reply:
@eric-wieser, thanks for tracking this issue down.  I found that it works without using @ and _ on the first argument, i.e. to take just `(M : general_linear_group n R)` rather than `(M : @general_linear_group n R ‹_› _ _)`.  OK with you?